File: backend/app/models/product.py
from tortoise import fields
from tortoise.models import Model


class Product(Model):
    id = fields.IntField(pk=True)
    name = fields.CharField(max_length=200)
    description = fields.TextField(null=True)
    price = fields.DecimalField(max_digits=10, decimal_places=2)
    stock_quantity = fields.IntField(default=0)
    created_at = fields.DatetimeField(auto_now_add=True)
    updated_at = fields.DatetimeField(auto_now=True)

    class Meta:
        table = "products"
        ordering = ["-created_at"]

File: frontend/src/types/product.ts
export interface Product {
  id: number
  name: string
  description: string | null
  price: string
  stock_quantity: number
  image_url: string | null
  created_at: string
  updated_at: string
}

export interface ProductInput {
  name: string
  description: string | null
  price: string
  stock_quantity: number
  image_url: string | null
}

export interface ProductPage {
  items: Product[]
  total: number
  page: number
  page_size: number
}
